iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -231,7 +231,16 @@ void RouDi::processMessage(const runtime::IpcMessage& message,
         }
         else
         {
-            capro::ServiceDescription service(cxx::Serialization(message.getElementAtIndex(2)));
+            auto maybeService =
+                capro::ServiceDescription::deserialize(cxx::Serialization(message.getElementAtIndex(2)));
+            if (maybeService.has_error())
+            {
+                LogError() << "Deserialization failed when '" << message.getElementAtIndex(2).c_str()
+                           << "' was provided\n";
+                break;
+            }
+            auto service = maybeService.value();
+
             cxx::Serialization portConfigInfoSerialization(message.getElementAtIndex(7));
 
             if (!service.isValid())
```

Comment:
Further down there is the de-serialization of the publisher and subscriber option but without error handling. Could you either also fix those o create and issue?

Reply:
@mossmaurice done with #989